cms-cat.spec

```diff
@@ -0,0 +1,42 @@
+### RPM cms cms-cat 240702.0
+## NOCOMPILER
+## NO_VERSION_SUFFIX
+
+# ***Do not change minor number of the above version. ***
+
+%define commit 0e3b60982ec088096fa67c46db5e9de2290ea5fb
+%define branch master
+# We do not use a revision explicitly, because revisioned packages do not get
+# updated automatically when there are dependencies.
+%define fakerevision %(echo %realversion | cut -d. -f1)
+Source0: git://gitlab.cern.ch/cms-analysis/services/cms.cern.ch-cat.git?obj=%{branch}/%{commit}&export=cms-cat&output=/cms-cat-%{commit}.tgz
+
+%prep
+%setup -n %{n}
+
+%build
+
+%install
+mkdir -p %{i}/cat
+mv * %{i}/cat
+
+%post
+cd ${RPM_INSTALL_PREFIX}/%{pkgrel}
+mkdir -p ${RPM_INSTALL_PREFIX}/cat ${RPM_INSTALL_PREFIX}/etc/%{pkgname}
+
+#Check if a newer revision is already installed
+if [ -f ${RPM_INSTALL_PREFIX}/etc/%{pkgname}/version ] ; then
+  oldrev=$(cat ${RPM_INSTALL_PREFIX}/etc/%{pkgname}/version )
+  if [ ${oldrev} -ge %{fakerevision} ] ; then
+    exit 0
+  fi
+fi
+
+for file in $(find . -name '*' -type f -path '*/cat/*') ; do
+  cp -f ${file} ${RPM_INSTALL_PREFIX}/${file}
```

Comment:
@clelange , looks like there is a directory under `cat`, so this command fails with
```
RPM installation stderr cms-cat:
cp: cannot create regular file '/tmp/muzaffar/w/./cat/combine_alias/exec_script.sh': No such file or directory
cp: cannot create regular file '/tmp/muzaffar/w/./cat/combine_alias/alias_script.sh': No such file or directory
```

If only thing which should go under `${RPM_INSTALL_PREFIX}/cat` is the `cat` directory then why not use
```
rsync -a --delete ${RPM_INSTALL_PREFIX}/%{pkgrel}/cat/ ${RPM_INSTALL_PREFIX}/cat/
```
this will allow to cleanup any old files which you might remove in new verions

Reply:
note that we should only use `--delete` here if only `cms-cat` copies its contents in `${RPM_INSTALL_PREFIX}/cat` . If you plan to have multiple packages copying their contents in `${RPM_INSTALL_PREFIX}/cat` then we should not use `--delete` option here

Reply:
OK, I see. I think we'll only deploy from a single repository for now, so I'll change to the suggest `rsync` command and will delete the rest.

What's the purpose of the second `cp` for-loop

```shell
for file in $(find . -name '*' -type l -path '*/cat/*') ; do
  cp -pRf ${file} ${RPM_INSTALL_PREFIX}/${file}
done
```

I guess this should also go? I'll remove it for now.